### DISCUSSION TITLE
HttpClient without a CORS enabled endpoint

### DISCUSSION THREAD

**Discussion by DmitryBorodiy** [unoplatform/uno#11104] [Q&A]
### Current behavior

Hello, Uno Platform community!
I try to get google search predictions with HttpClient.

string _suggestSearchUrl = "http://www.google.com/complete/search?output=toolbar&q={0}&hl=en";
strring query = "Example search text";

using(HttpClient client = new HttpClient())
{
         result = await client.GetStringAsync(String.Format(_suggestSearchUrl, query));
}

This code works great on UWP and Android, but doesn't work on WASM.
Thanks!

### Expected behavior

This code should work as if it were on the UWP.

### Workaround

HttpClient using to get search predictions from Google.

**Comment by jeromelaban** [comment]
Thanks for the question. If you open the browser debugger, you'll notice the following message:
```
Access to fetch at 'https://www.google.com/complete/search?output=toolbar&q=Example%20search%20text&hl=en'
 (redirected from 'http://www.google.com/complete/search?output=toolbar&q=Example%20search%20text&hl=en') 
from origin 'http://localhost:5000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header
is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors'
to fetch the resource with CORS disabled.
```

This is an expected behavior, see [this document](https://platform.uno/docs/articles/get-started-wizard.html?#webassembly-access-to-fetch-at-httpsxxxx-from-origin-httpxxxx-has-been-blocked-by-cors-policy).

We're currently updating our packages to enable `no-cors` fetches, we'll update this thread when it becomes available.

**Reply by DmitryBorodiy** [reply]
When will this approximately work and what needs to be done for the current project to fix it in the future?

**Reply by jeromelaban** [reply]
The package `Uno.Wasm.HttpRequestMessageExtensions` (7.0.16) is now available, but it will not fix your scenario, as [it is a browser requirement](https://stackoverflow.com/a/43268098/26346). `no-cors` will only prevent the request to fail, AND you will not get the response body (see https://github.com/dotnet/aspnetcore/issues/24080#issuecomment-661184383 for a .NET specific discussion). 

There's no solution to read the response without the server specifying CORS heads, or by using separate service (like a specialized Azure function) which bypasses CORS requests.

If you need additional explanations about CORS and `no-cors` see https://evertpot.com/no-cors.
